cartographer/mapping_2d/pose_graph/landmark_cost_function.h

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2018 The Cartographer Authors
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef CARTOGRAPHER_MAPPING_2D_POSE_GRAPH_LANDMARK_COST_FUNCTION_H_
+#define CARTOGRAPHER_MAPPING_2D_POSE_GRAPH_LANDMARK_COST_FUNCTION_H_
+
+#include "cartographer/mapping/pose_graph.h"
+#include "cartographer/mapping_2d/pose_graph/spa_cost_function.h"
+#include "cartographer/transform/rigid_transform.h"
+#include "cartographer/transform/transform.h"
+#include "ceres/ceres.h"
+#include "ceres/jet.h"
+
+namespace cartographer {
+namespace mapping_2d {
+namespace pose_graph {
+
+// Cost function measuring weighted error between the observed relative pose
```

Comment:
...the weighted error between the observed pose given by the landmark measurement and the linearly interpolated pose...
(same in 3D)

Reply:
Done